### PR TITLE
GoSec Security Scan Issues Resolved

### PR DESCRIPTION
**Pull request by ChristianAtDell**
# Description
Six GoSec security scan issues were previously detected, primarily unsanitized filename inputs. I have resolved five of them, done a thorough investigation of the sixth, and determined that it is safe to add a \#nosec to it because one of its two inputs MUST be variable but it is a program-generated input that is not at risk of data injection. 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| [632](https://github.com/dell/csm/issues/632) |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

File: test/mock-server/stub/stub.go [hunk 1]
```diff
@@ -62,7 +63,11 @@ func RunStubServer(opt Options) {
 
 	fmt.Println("Serving stub admin on http://" + addr)
 	go func() {
-		err := http.ListenAndServe(addr, r)
```
Comment (santhoshatdell) [inline test/mock-server/stub/stub.go:65]: 'r' var is not used in the new lines of code. Please check.

**Review by santhoshatdell — COMMENTED** [review]
We also need to update the workflow action yaml file.
Please refer https://github.com/dell/csm-replication/commit/952a1551725bc13db8e0aef2fbcb3143f1ebe8a9

**Comment by ChristianAtDell** [comment]
> We also need to update the workflow action yaml file. Please refer [952a155](https://github.com/dell/csm-replication/commit/952a1551725bc13db8e0aef2fbcb3143f1ebe8a9)

I see that they use ./.. instead of ./...

In my experience, ./... is necessary to scan subfolders. 

gosec ./.. output summary: 

```
  Gosec  : dev
  Files  : 0
  Lines  : 0
  Nosec  : 0
  Issues : 0
```

gosec ./... output summary: 

```
  Gosec  : dev
  Files  : 63
  Lines  : 13684
  Nosec  : 20
  Issues : 0
```

**Review by santhoshatdell — DISMISSED** [review]

**Comment by ChristianAtDell** [comment]
Sorry for the e-mail spam, everyone. I didn't realize it was sending them out until after many commits. 

Should be ready for review now, though.

**Review by santhoshatdell — APPROVED** [review]

**Review by anandrajak1 — APPROVED** [review]

**Pull request merged**